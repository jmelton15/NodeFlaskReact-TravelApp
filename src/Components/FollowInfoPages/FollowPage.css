.FollowPage-ConnectionsListContainer {
  margin-top: 20px;
  width:40% !important;
}

@media only screen and (min-device-width: 5px) and (max-device-width: 420px) {
  .FollowPage-ConnectionsListContainer {
    margin-top: 20px;
    width:100% !important;
  }
}