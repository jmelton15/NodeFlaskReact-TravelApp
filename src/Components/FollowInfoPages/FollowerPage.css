.FollowerPage-ConnectionsListContainer {
    margin-top: 20px;
    width:40% !important;
}

  
@media only screen and (min-device-width: 5px) and (max-device-width: 420px) {
  .FollowerPage-ConnectionsListContainer {
    margin-top: 20px;
    width:100% !important;
  }
}