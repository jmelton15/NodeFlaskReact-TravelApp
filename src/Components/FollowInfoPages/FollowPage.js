import "./FollowPage.css";
import ConnectionCard from "../AddConnections/ConnectionCard";
import { Redirect } from "react-router-dom";
import GetScreenWidth from '../../helpers/GetScreenWidth';

const FollowPage = ({user,token}) => {
    if(!token || !user) {
        return <Redirect to="/"></Redirect>
    }

    const [screenWidth] = GetScreenWidth();
    return (
        <div className={screenWidth > 600 ? "FollowPage-ConnectionsListContainer container" : "FollowPage-ConnectionsListContainer px-3"}>
            {user.following.map((connection) => {
                return <ConnectionCard connection={connection} user={user} token={token} />
            })}
        </div>
    )
}

export default FollowPage;