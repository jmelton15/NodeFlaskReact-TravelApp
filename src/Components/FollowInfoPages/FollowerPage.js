import "./FollowerPage.css";
import ConnectionCard from "../AddConnections/ConnectionCard";
import { Redirect } from "react-router-dom";
import GetScreenWidth from '../../helpers/GetScreenWidth';

const FollowerPage = ({user,token}) => {
    if(!token || !user) {
        return <Redirect to="/"></Redirect>
    }
    const [screenWidth] = GetScreenWidth();
    
    return (
        <div className={screenWidth > 600 ? "FollowerPage-ConnectionsListContainer container" : "FollowerPage-ConnectionsListContainer px-3"}>
            {user.followers.map((connection) => {
                return <ConnectionCard connection={connection} user={user} token={token} />
            })}
        </div>
    )
}

export default FollowerPage;