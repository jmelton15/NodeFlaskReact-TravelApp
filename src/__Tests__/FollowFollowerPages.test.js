import { fireEvent, render } from '@testing-library/react';
import {user,connection} from "./TestHelperObjects";
import FollowerPage from '../Components/FollowInfoPages/FollowerPage';
import FollowPage from '../Components/FollowInfoPages/FollowPage';

// simple smoke test for FollowerPage component
it("renders a FollowerPage component without crashing", function() {
    render(<FollowerPage user={user} token={true} />);
});

// simple snapshot test for v Component
it("should match snapshot for FollowerPage", function() {
    const {asFragment} = render (<FollowerPage user={user} token={true} />);
    expect(asFragment()).toMatchSnapshot();
});

// simple smoke test for FollowPage component
it("renders a FollowPage component without crashing", function() {
    render(<FollowPage user={user} token={true} />);
});

// simple snapshot test for FollowPage Component
it("should match snapshot for FollowPage", function() {
    const {asFragment} = render (<FollowPage user={user} token={true} />);
    expect(asFragment()).toMatchSnapshot();
});
