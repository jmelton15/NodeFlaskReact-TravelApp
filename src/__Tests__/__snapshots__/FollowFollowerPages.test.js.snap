// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for FollowPage 1`] = `
<DocumentFragment>
  <div
    class="FollowPage-ConnectionsListContainer container"
  >
    <div
      class="ConnectionCard-Card card"
    >
      <img
        alt="Avatar Profile Pic"
        class="card-img"
        id="connectioncard-avatarimg"
        src="some pic route"
      />
      <div
        class="card-body"
      >
        <div
          id="connectioncard-usernameBio"
        >
          <h1
            class="text-center card-title"
            id="userinfocard-username"
          >
            TESTCONN
          </h1>
          <h6
            class="mb-2 text-muted card-subtitle"
          >
            Connection To Test1
          </h6>
        </div>
        <p
          class="card-text"
          id="connectioncard-followingInfo"
        />
        <p
          id="connectioncard-followers"
        >
          <span>
            Followers:
          </span>
           1
        </p>
        <p
          id="connectioncard-following"
        >
          <span>
            Following:
          </span>
           1
        </p>
        <p />
      </div>
      <div
        class="ConnectionCard-ButtonContainer"
      >
        <blockquote
          id="alreadyFollowing"
        >
          (Already Following)
        </blockquote>
        <button
          class="btn btn-secondary"
          id="connectioncard-followbtn"
          type="button"
        >
          UnFollow
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot for FollowerPage 1`] = `
<DocumentFragment>
  <div
    class="FollowerPage-ConnectionsListContainer container"
  >
    <div
      class="ConnectionCard-Card card"
    >
      <img
        alt="Avatar Profile Pic"
        class="card-img"
        id="connectioncard-avatarimg"
        src="some pic route"
      />
      <div
        class="card-body"
      >
        <div
          id="connectioncard-usernameBio"
        >
          <h1
            class="text-center card-title"
            id="userinfocard-username"
          >
            TESTCONN
          </h1>
          <h6
            class="mb-2 text-muted card-subtitle"
          >
            Connection To Test1
          </h6>
        </div>
        <p
          class="card-text"
          id="connectioncard-followingInfo"
        />
        <p
          id="connectioncard-followers"
        >
          <span>
            Followers:
          </span>
           1
        </p>
        <p
          id="connectioncard-following"
        >
          <span>
            Following:
          </span>
           1
        </p>
        <p />
      </div>
      <div
        class="ConnectionCard-ButtonContainer"
      >
        <blockquote
          id="alreadyFollowing"
        >
          (Already Following)
        </blockquote>
        <button
          class="btn btn-secondary"
          id="connectioncard-followbtn"
          type="button"
        >
          UnFollow
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
